js/colourpicker.js

```diff
@@ -153,16 +143,22 @@
       if( input.data('colourpicker-initialized') ) return;
 
       // Handle settings
-      settings = $.extend(true, {}, defaults, settings);
-
-	    // Palette type
-	    colourpicker.addClass('palette-' + settings.palette);
-	  
-	    // If adding a transparent checkbox, make the parent an input group
-      if( settings.allowTransparent ) {
-        colourpicker.addClass('input-group');
+      settings = $.extend(false, {}, defaults, settings);
+      //for backwards compatibility
+      if ($.type(settings.allowedCols) === "string") {
```

Comment:
This is generally a great idea, it's good that you kept backwards compatibility in mind. But in this very specific case I don't think we need to do that because this library is literally only used for the shiny package, which will know to send in the correct format when we use this updated version. I highly doubt anybody else is using this library. If you think someone might be using it and would be affected, then keep this check,